experiments/compositions/composition/tests/testclient/client.go

```diff
@@ -0,0 +1,323 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kustomize/kstatus/status"
+)
+
+var scheme = runtime.NewScheme()
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(compositionv1.AddToScheme(scheme))
+}
+
+// Client - test client of MCS target cluster
+type Client struct {
+	T   *testing.T
+	Ctx context.Context
+	cs  *kubernetes.Clientset
+	client.Client
+	name     string
+	testName string
+}
+
+func New(ctx context.Context, t *testing.T, config *rest.Config, name, testName string) *Client {
+	c, err := client.New(config, client.Options{Scheme: scheme})
+	require.NoError(t, err)
+	clientset, err := kubernetes.NewForConfig(config)
+	require.NoError(t, err)
+	return &Client{
+		T:        t,
+		Ctx:      ctx,
+		cs:       clientset,
+		Client:   c,
+		name:     name,
+		testName: testName,
+	}
+}
+
+func (c *Client) String() string {
+	return c.name
+}
+
+func (c *Client) WriteNamespacePodLogs(namespace string) {
+	// Get pods in namespace
+	pods, err := c.cs.CoreV1().Pods(namespace).List(c.Ctx, metav1.ListOptions{})
+	require.NoError(c.T, err)
+
+	for _, pod := range pods.Items {
+		// Get containers in pod
+		for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
+			// Write container logs
+			c.WriteContainerLogs(namespace, pod.Name, container.Name)
+		}
+	}
+}
+
+func (c *Client) getFilePath(namespace, name, container string) string {
+	path := filepath.Join("test-logs", c.testName, namespace)
+	os.MkdirAll(path, os.ModePerm)
+	filename := name + "." + container + ".txt"
+	return filepath.Join("test-logs", c.testName, namespace, filename)
+}
+
+func (c *Client) ClearOldLogs() {
+	path := filepath.Join("test-logs", c.testName)
+	os.RemoveAll(path)
+}
+
+func (c *Client) WriteContainerLogs(namespace, name, container string) {
+	podLogOpts := corev1.PodLogOptions{Container: container}
+
+	req := c.cs.CoreV1().Pods(namespace).GetLogs(name, &podLogOpts)
+
+	podLogs, err := req.Stream(c.Ctx)
+	defer podLogs.Close()
+	require.NoError(c.T, err)
+
+	file, err := os.Create(c.getFilePath(namespace, name, container))
+	defer file.Close()
+	require.NoError(c.T, err)
+	_, err = io.Copy(file, podLogs)
+	require.NoError(c.T, err)
+}
+
+// MustCreate - create object
+func (c *Client) MustCreate(u *unstructured.Unstructured) {
+	c.T.Helper()
+
+	id := ExtractGVKNN(u)
+	c.T.Logf("Creating object %q on cluster %q", id, c)
+	err := c.Create(c.Ctx, u)
+	if err != nil {
+		exists := apierrors.IsAlreadyExists(err)
+		require.Truef(c.T, exists, "failed to create absent %q: %s", id, err)
```

Comment:
Similarly this can just be an if statement I think; that's why `require` is generally considered not-go

Reply:
replaced with if stt. removed usage of `require`